libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -347,10 +361,12 @@ void DataFlowAnalyzer::clearValues(std::set<YulName> _variables)
 
 	// Also clear variables that reference variables to be cleared.
 	std::set<YulName> referencingVariables;
-	for (auto const& variableToClear: _variables)
-		for (auto const& [ref, names]: m_state.references)
-			if (names.count(variableToClear))
-				referencingVariables.emplace(ref);
+	std::vector const varVec(_variables.begin(), _variables.end());
+	for (auto const& [ref, names]: m_state.references)
```

Comment:
These names could use some more detail. What I really want to know when I read this is how they differ from each other:

```suggestion
	std::vector const sortedVariablesToClear(_variablesToClear.begin(), _variablesToClear.end());
	for (auto const& [referencingVariable, referencedVariables]: m_state.references)
```

Though now the `referencingVariables` set starts to sound like a set of all 
referencing variables so I'd rename it too: `referencingVariablesToClear`.

Reply:
Yeah makes sense. Fixed!